Sources/SwiftDocC/Infrastructure/External Data/OutOfProcessReferenceResolver.swift

```diff
@@ -762,30 +810,54 @@ extension OutOfProcessReferenceResolver {
+        }
+        
+        var kindVariants = [mainTrait: symbolKind(forNodeKind: kind)]
+        var titleVariants = [mainTrait: title]
+        var subHeadingVariants = [DocumentationDataVariantsTrait: [SymbolGraph.Symbol.DeclarationFragments.Fragment]]()
+        subHeadingVariants[mainTrait] = declarationFragments?.declarationFragments
+        var abstractVariants = [DocumentationDataVariantsTrait: AbstractSection]()
+        abstractVariants[mainTrait] = abstract.map { AbstractSection(paragraph: Paragraph([Text($0)])) }
+        
+        for variant in variants ?? [] {
+            guard case .interfaceLanguage(let language) = variant.traits.first else { continue }
+            let trait = DocumentationDataVariantsTrait(interfaceLanguage: language)
+            
+            if let variantKind = variant.kind {
+                kindVariants[trait] = symbolKind(forNodeKind: variantKind)
+            }
+            if let variantTitle = variant.title {
+                titleVariants[trait] = variantTitle
+            }
+            if let variantAbstract = variant.abstract {
+                abstractVariants[trait] = AbstractSection(paragraph: Paragraph([Text(variantAbstract)]))
+            }
+            if let variantDeclarationFragments = variant.declarationFragments {
+                subHeadingVariants[trait] = variantDeclarationFragments?.declarationFragments
+            }
         }
         
         return Symbol(
-            kindVariants: .init(swiftVariant: symbolKind(forNodeKind: kind)),
-            titleVariants: .init(swiftVariant: title),
-            subHeadingVariants: .init(swiftVariant: declarationFragments?.declarationFragments),
-            navigatorVariants: .init(swiftVariant: nil),
-            roleHeadingVariants: .init(swiftVariant: ""), // This information isn't used anywhere since this node doesn't have its own page, it's just referenced from other pages.
-            platformNameVariants: .init(swiftVariant: nil),
-            moduleNameVariants: .init(swiftVariant: ""), // This information isn't used anywhere since the `urlForResolvedReference(reference:)` specifies the URL for this node.
-            externalIDVariants: .init(swiftVariant: nil),
-            accessLevelVariants: .init(swiftVariant: nil),
-            availabilityVariants: .init(swiftVariant: availability),
-            deprecatedSummaryVariants: .init(swiftVariant: nil),
-            mixinsVariants: .init(swiftVariant: nil),
-            abstractSectionVariants: .init(swiftVariant: nil),
-            discussionVariants: .init(swiftVariant: nil),
-            topicsVariants: .init(swiftVariant: nil),
-            seeAlsoVariants: .init(swiftVariant: nil),
-            returnsSectionVariants: .init(swiftVariant: nil),
-            parametersSectionVariants: .init(swiftVariant: nil),
-            redirectsVariants: .init(swiftVariant: nil)
+            kindVariants: .init(values: kindVariants, defaultVariantValue: nil),
```

Comment:
I noticed we _do_ have a default value here so we could make these a little more consise if you like. But I see the benefit in being explicit as well.

```suggestion
            kindVariants: .init(values: kindVariants),
```